mixins/localize-mixin.js

```diff
@@ -192,3 +194,7 @@ export const LocalizeMixin = dedupeMixin(superclass => class LocalizeMixinClass
 	}
 
 });
+
+export const linkGenerator = ({ href, target }) => {
```

Comment:
Yeah... a couple options:

1. We could do a dynamic import in here anyway (I pushed that up as an option). Since it's async it'll fire in the background, so it means it's not guaranteed to be loaded before the result is rendered. FYI this is what the earlier implementation of `localizeHTML` was doing already.
2. We could put these helpers in either a single `localize-helpers.js` file (which imports all relevant components, a bit inefficient), or each helper into its own separate file to truly isolate the imports.

Reply:
I'm slightly less concerned if we can provide a factory to produce replacement handlers.

It is possible to do async rendering, but I don't think we would really want to because I think it would complicate consumer code quite a bit. I'd rather just import those couple components. In practice it's not amazing, but in reality we're not talking about much (yeah, could be a slippery slope), and there's a good chance it could be loaded anyway. 

If we are worried about it, we could look at ways to factor out this `localizeHTML` stuff into a separate module that the consumer has to import, so that we're only importing these dependencies when they are likely to be needed. I really wish all this localization stuff wasn't tied to components.

Reply:
FOUC I think?

Reply:
Yeah not just FOUC (unstyled), but nothing at all will be rendered until the component is registered. Could be a headache for code that expects the component to be ready, e.g. visual-diff tests.

Reply:
And then the downstream consequence of FOUC/deferred rendering being more repaints/reflows later in the render cycle.

Unless we see evidence to the contrary, I'm not too worried about this, since loading the localization resources itself is also `async`, so there's already deferred rendering going on.

Reply:
So @dlockhart, you think dynamic imports are fine? I could see it either way, static imports would be easy enough especially if the helpers were in their own file.

On that note:

1. If we're providing helpers, I think putting them in their own file like `localize-helpers.js` would be better than in `localize-mixin.js` (notice the consumer isn't actually using `localize-mixin.js` for the mixin itself anyway)

2. No opinion on what the ideal naming for the helpers would be (`linkGenerator`, `linkFactory`, `linkLocalizeHelper`, ...)

3. I also think with this approach, `localizeHTML()` is no longer necessary, it can all be part of `localize()` directly (especially since consumers can get the same behaviour directly out of `localize()` anyway)

Reply:
I'm not sure I love exposing all these generators and requiring their explicit use. As a consumer, I'd rather just pass the info to `localize`/`localizeHTML` and expect it to handle the generation however is necessary internally. Like, the result of the generator isn't really useful to me, it's only useful to the thing I'm guaranteed to already be sending it to.

```javascript
const links = {
	link1:  { href: 'https://wikipedia.org/wiki/Culture_change', target: '_blank' },
	link2: { href: 'https://wikipedia.org/wiki/Earth', target: '_blank' }
];
return html`
	${this.localizeHTML('mission', {...}, { links })}
`;
```

```javascript
localizeHTML(key, replacements, links) {
	...
	const _links = Object.entries(links).reduce((acc, [k, v]) => acc[k] = linkGenerator(v) && acc, {});
	this.localize(key, { ...replacements, ..._links });
}

Or, just back to writing the html myself, but _that_ brings us back to how much we want to control this. We could filter out `replacements` that are functions too. Then I also wonder if we're just babying people too much. I guess I'm just all over the place and would be happy to just be able to do this in any manner 😆